src/Microsoft.AspNetCore.OData/Batch/ODataBatchHttpRequestMessageExtensions.cs

```diff
@@ -52,7 +54,8 @@ public static bool IsODataBatchRequest(this HttpRequest request)
                 throw Error.ArgumentNull("request");
             }
 
-            return request.ContentType?.StartsWith("multipart/", StringComparison.OrdinalIgnoreCase) ?? false;
+            return (request.ContentType != null &&
+                (request.ContentType.StartsWith(BatchMediaTypeMime) || request.ContentType.StartsWith(BatchMediaTypeJson)));
```

Comment:
>StartsWith [](start = 37, length = 10)

maybe compare as equality?

Reply:
We need to use StartWith here -- ContentType header value typically consists of type + parameter values, therefore Equality won't work.

---
In reply to: [202867652](https://github.com/OData/WebApi/pull/1536#discussion_r202867652) [](ancestors = 202867652)